### PR TITLE
Port-DirectoryContents

### PR DESCRIPTION
**Pull request by MakoEnergy**
Initial commit for Directory Contents.

### CODE REVIEW

File: src/DirectoryContents.cpp [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+   you have any questions.
+
+   Joseph Toppi - toppij@gmail.com
+   John Blackwood - makoenergy02@gmail.com
+*/
+
+#include "CrossPlatformExport.h"
+
+#ifdef MEZZ_Windows
+    #define _WIN32_WINNT 0x0601
```
Comment (Sqeaky) [inline src/DirectoryContents.cpp:45]: I am pretty sure this just sets some specific version of windows, but that probably should be commented.

File: src/DirectoryContents.cpp [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+
+#ifdef MEZZ_Windows
+    /// @brief Convenience type because Windows likes fatty Strings.
+    using WideString = std::wstring;
+
+    /// @brief Demotes a size_t to an int, which is required by the Win API.
+    /// @brief ToDemote The large integral type to be demoted.
+    /// @return Returns an int of the same value if it can be stored, or asserts otherwise.
+    [[nodiscard]]
+    int IntDemote(const size_t ToDemote)
```
Comment (Sqeaky) [inline src/DirectoryContents.cpp:78]: This has more general purpose use and might be put in Foundation or StaticFoundation.

File: src/DirectoryContents.cpp [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+    #endif
+        return Ret;
+    }
+
+    ArchiveEntryVector GetDirectoryContents(const StringView DirectoryPath)
+    {
+        ArchiveEntryVector Ret;
+    #ifdef MEZZ_Windows
+        WIN32_FIND_DATAW FileData;
+        //LARGE_INTEGER FileSizeConverter;
```
Comment (Sqeaky) [inline src/DirectoryContents.cpp]: Leftovers?

File: src/DirectoryContents.cpp (hunk 1 above)
Comment (MakoEnergy) [reply src/DirectoryContents.cpp:45]: Documented.

File: src/DirectoryContents.cpp (hunk 2 above)
Comment (MakoEnergy) [reply src/DirectoryContents.cpp:78]: Issue for this has been created. #4

File: src/DirectoryContents.cpp (hunk 3 above)
Comment (MakoEnergy) [reply src/DirectoryContents.cpp]: Removed.

**Pull request merged**